### PR TITLE
feat: added fair launch claim

### PR DESCRIPTION
**Pull request by zhyd1997**
Close: #101

### CODE REVIEW

**Comment by zhyd1997** [comment]
Hi, @codynhat 

Why did I get `_auctionStart` and `_auctionEnd` both `0`:
```js
  const _auctionStart = await claimerContract.auctionStart(); // 0
  const _auctionEnd = await claimerContract.auctionEnd();     // 0
```

**Comment by zhyd1997** [comment]
> The current deployment has them set to 0. But we can change it. Do you need me to change it for testing?

Got it, no need to change it, I can handle this situation.

**Comment by zhyd1997** [comment]
Hi, @codynhat 

2 questions:

### a. what's the `block` here:

https://github.com/Geo-Web-Project/core-contracts/blob/bb6b1d0e12c03615e1ab21dc7d5e6d19abfb9cc4/contracts/FairLaunchClaimer.sol#L84

### b. and the `Claim` action needs to do the **same** thing with **Simple Claim View**?

File: components/Sidebar.tsx [hunk 1]
```diff
@@ -93,22 +95,81 @@ function Sidebar(props: SidebarProps) {
...
-        pinningManager={pinningManager}
-        licenseAddress={licenseContract.address}
-      ></ParcelInfo>
+      {isFairLaunch ? (
+        <FairLaunchInfo
+          currentRequiredBid={truncateEth(
+            ethers.utils.formatEther(requiredBid),
+            4
+          )}
+          auctionEnd="32376137040000"
```
Comment (codynhat) [inline components/Sidebar.tsx:159]: Should this be `auctionEnd`?

File: components/Sidebar.tsx [hunk 2]
```diff
@@ -93,22 +95,81 @@ function Sidebar(props: SidebarProps) {
...
+    endingBid &&
+    Date.now() > auctionStart.toNumber() &&
+    Date.now() < auctionEnd.toNumber();
+
+  const requiredBid =
+    auctionStart && auctionEnd && startingBid && endingBid
+      ? calculateRequiredBid(auctionStart, auctionEnd, startingBid, endingBid)
+      : "0";
+
+  console.log("requiredBid: ------->: ", requiredBid);
```
Comment (codynhat) [inline components/Sidebar.tsx:145]: Can you remove the console.log before merging?

File: components/cards/FairLaunchInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      </Row>
+      <Card border="secondary" className="bg-dark my-5">
+        <Card.Body>
+          <Card.Title className="text-primary font-weight-bold">
+            Auction Details
+          </Card.Title>
+          <Card.Text>Current Required Bid: {currentRequiredBid} ETHx</Card.Text>
+          <Card.Text>Auction End: {formattedAuctionEnd} UTC</Card.Text>
+          <Card.Text>
+            Time Remaining: <span>{duration.days()}</span>:
```
Comment (codynhat) [inline components/cards/FairLaunchInfo.tsx:64]: Is there a way to add zeros for padding?

For example, the time currently shows `5:7:17:8`. Could this be -> `5:07:17:08`?

**Review by codynhat — CHANGES_REQUESTED** [review]

**Review by codynhat — APPROVED** [review]
LGTM!

**Pull request merged**